fix: Fix `asyncLoader` type signature

shun-shobon opened the pull request:
### Why

resolves #309 

### What

- Remove `loadAsync` definition from `Loader`.
- Add `loadAsync` definition to some loaders.
- Fix type signature to `loadAsync` from `CubeTextureLoader` and `HDRCubeTextureLoader`.

### Checklist

-   [x] Checked the target branch (current goes `master`, next goes `dev`)
-   [x] Added myself to contributors table
-   [x] Ready to be merged

shun-shobon committed:
fix: Fix `asyncLoader` type signature

shun-shobon committed:
fix: Change argument name

shun-shobon committed:
docs: Add me to contributors

shun-shobon commented:
I found where `load` and `loadAsync` have different return types. Should I fix it?
https://github.com/three-types/three-ts-types/blob/3ad9711aa8d61f9a5ea0c1a6a89d66c65fc46d84/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18

0b5vr commented:
> I found where `load` and `loadAsync` have different return types. Should I fix it?

Blame: https://github.com/three-types/three-ts-types/blame/3ad9711aa8d61f9a5ea0c1a6a89d66c65fc46d84/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18

We probably forgot to change it when we support `VOXMesh` ? I agree to change the return type to `Promise<Chunk[]>`.

0b5vr commented on .all-contributorsrc:
```diff
@@ -494,6 +494,16 @@
+      "avatar_url": "https://avatars.githubusercontent.com/u/37236438?v=4",
+      "profile": "https://shun.technology/",
+      "contributions": [
+        "bug",
```
correct! most of the people seems to forget this one 😄

0b5vr commented:
Thanks! Approved for now but if you could the `VOXLoader` fix we've discussed it would be awesome.

shun-shobon committed:
fix: Fix return type to `Promise<Chunk[]>`

shun-shobon committed:
docs: Update my profile URL

shun-shobon commented:
> > I found where `load` and `loadAsync` have different return types. Should I fix it?
> 
> Blame: [`3ad9711`/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18 (blame)](https://github.com/three-types/three-ts-types/blame/3ad9711aa8d61f9a5ea0c1a6a89d66c65fc46d84/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18)
> 
> We probably forgot to change it when we support `VOXMesh` ? I agree to change the return type to `Promise<Chunk[]>`.

OK. I changed the return type 😄 (and changed my profile URL from the old one)

0b5vr commented:
Thanks!

0b5vr commented:
oof I overlooked the target branch. The changes corresponding to the current Three.js versions should have been directed to `master`, not `dev`. It's gonna be released in the next version (r149) anyway so don't worry

shun-shobon commented:
I should have directed it to master, I misunderstood that it had to be directed to dev. I'm sorry...
